src/util/ExpressionUtils.h

```diff
@@ -101,6 +101,10 @@ class ExpressionUtils {
         Expression::Kind kind,
         const std::vector<std::unique_ptr<Expression>>& rels);
 
+    static Expression* flattenInnerLogicalAndExpr(const Expression* expr);
+    static Expression* flattenInnerLogicalOrExpr(const Expression* expr);
+    static Expression* flattenInnerLogicalExpr(const Expression* expr);
```

Comment:
Return std::unique_ptr;

Don't let user of these functions free the returned memory.

Reply:
ACK.